### DISCUSSION TITLE
API 33

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#2775] [Q&A]
Hi all, 

I received an email from Google saying that "We have detected that 4 of your apps target an older version of Android. To provide a secure experience for users, Google Play requires all apps to meet the requirements for the target API level by August 31, 2023."
It seems that we have to target **API 33**. 
What I don't understand is that, for my builds, I'm using the last AIR version, and specifed in the xml : 
<uses-sdk android:minSdkVersion="28" android:targetSdkVersion="33" />	

And my build, on Google Play store, indicates "API level : 16 or higher" "Target SDK : 33"
So, what Google is asking me ? 

Thank you for your answers

![Capture d'écran 2023-08-19 133819](https://github.com/airsdk/Adobe-Runtime-Support/assets/6382334/6b842313-4088-4a01-9aa8-86b0f508c341)

**Comment by marchbold** [comment]
Simply add the following to the manifest additions in your application descriptor:

```xml
<uses-sdk android:minSdkVersion="19" android:targetSdkVersion="33"/>
```

eg: 
```xml
<android>
  <manifestAdditions><![CDATA[
    <manifest android:installLocation="auto" >
      <uses-sdk android:minSdkVersion="19" android:targetSdkVersion="33"/>

      <!-- OTHER MANIFEST ADDITIONS -->

    </manifest>
  ]]></manifestAdditions>
</android>
```

**Reply by stephanberger4** [reply]
I did.. That's why it's weird
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6382334/43429572-cc33-41df-8b70-19e7e225c9a2)

**Reply by marchbold** [reply]
It's in the wrong place. Look at where I placed it above. Needs to be inside the `manifest` node in the manifest additions.

**Comment by FliplineStudios** [comment]
Also check if you have any older versions of APKs/AABs in a Closed Testing or Internal Testing track.  For us all of our production builds are targeting 33, but in the Policy status section we got similar warnings, and it would list the affected builds -- all of which were for non-production builds that were hanging out in testing tracks (even if those tracks are inactive, Google will still flag them as violations).

**Reply by marchbold** [reply]
Yeah good point! 

Also if you have old versions supporting devices that aren't supported by your latest build this can also affect review. Good idea to make sure you remove old apks / aabs as part of your update. Google doesn't really make this clear.

**Reply by stephanberger4** [reply]
Thanks for your answers. I've got this... (no difference), but does it mean my builds are with API 33 ?
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6382334/f57b37d2-f22b-42a8-a7c6-6583f170740c) :  "Target SDK : 33"

**Reply by stephanberger4** [reply]
Correct  ! It was the Beta version (API 31). THANK YOU !!

**Reply by FliplineStudios** [reply]
Glad that was it!  

For us we updated our testing track versions so they matched the production build with API level 33, but that didn't remove the policy warning (unless there's a delay between going live and Google realizing it changed...).  In the "How to fix" part of the policy message they mention that you must upload a new build to Production to remove the violation, which is odd especially if the Production build wasn't the problem in the first place.

**Reply by Michoko92** [reply]
I'm not sure how to completely remove the contents of testing tracks in the Google console. I'd like to clear them, but I only found the Pause Track button. Is there a way to remove all the APKs from a track please?

**Comment by Michoko92** [comment]
I actually got the same warning today. Some months ago I tediously updated all my apps to API 31, and I thought that for API 33, only apps updates or new apps would actually require API version change. But the message from Google feels like it is a mandatory upgrade before end of August.

Do you know if I can leave my apps with API 31, or is API 33 required for apps to remain available on the Play Store? Thank you!

**Reply by ajwfrost** [reply]
In our console view it says:
> From 31 Aug 2023, if your target API level is not within one year of the latest Android release, you won't be able to update your app.

So it sounds like you won't be able to update the app, but it would still be available. But I thought I'd read somewhere that they wouldn't show it to people who've got latest Android software running..  Ah yes:
> Existing apps must target API level 31 or above to remain available to users on devices running Android OS higher than your app's target API level. Apps that target API level 30 or below will only be available on devices running Android OS the same or lower than your apps’ target API level.

https://support.google.com/googleplay/android-developer/answer/11926878

Not sure how well it's working in terms of recognizing when you've done an update. We just published a new version of the Scout mobile config helper to update it to targetSDK 33, it's live now but they're still showing it as a policy violation even though they also say it's got the right target SDK API level...

**Reply by ageekhere** [reply]
I have also updated my apps to target SDK 33 and still have the warning, if I still have the warning with 1 day to go I think I am going to click the request more time button.

**Reply by stephanberger4** [reply]
I've also checked, seeing the previous messages, and false alert, my last build didn't solve the thing. Google is still asking that "my app must target Android 13 (API level 33)"...

Is it a bug with Google ? Or something to solve with AIR SDK ? 

Do we have to click on "request more time" ?

**Reply by FliplineStudios** [reply]
This morning I got Play Console messages in the Inbox titled "A recent policy warning about your app has been addressed" for each, with the policy status clear for all of our affected apps now.  I updated them all Friday morning and the updates went live Friday afternoon, so it may just take a little time for them to (manually?) register that the updates have been made.

**Reply by ageekhere** [reply]
The warning for my apps are now gone, so it did take a few days for it to clear for me.

**Reply by stephanberger4** [reply]
Would you mind uploading a printscreen of your App bundles just to check please ? Here's mine : 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6382334/8ac1ef56-231f-4c97-89fd-3c7345166e95)

**Reply by FliplineStudios** [reply]
@stephanberger4 Sure, here's mine:
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/33292459/1f76b160-c79f-4b9c-b078-eaa9607c7b01)

I had to update the Testing tracks for our apps, so make sure you don't have any live tracks that have older builds in them targeting a different SDK.

@Michoko92 I was really surprised by the warning too, since they're making it sound like you won't ever be able to submit an update if you're using API <33??  I don't remember that _ever_ being communicated with their API requirements before.  Previously they said that apps targeting API 31 would still be available for users, and any future updates would need to be API 33+, but they never said that having apps remaining on API 31 would prevent you from updating in the future...  until that showed up in this policy message.

**Reply by Michoko92** [reply]
Thank you all for your replies. @ajwfrost  this is exactly the information I was looking for, cheers! I really don't think it makes any sense to prevent apps from being updated to most recent SDK if they are still on the store. The way I understood this message is that you'd not be able to update the app as long the update doesn't target API 33. Do someone has a final word about this?

**Comment by raresn** [comment]
We have:
```
<manifest android:installLocation="auto">
<uses-sdk android:minSdkVersion="15" android:targetSdkVersion="33"/>
```
 yet we still get that messsage...
Does anyone else still get it?


![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/1768694/cfe2e563-d9e9-48b4-9d50-be5747cda455)

LE: 1 hour after pausing those tracks the message is gone :)
